File: python/beginners_selection/abc086c.py
# https://atcoder.jp/contests/abs/tasks/arc089_a

n = int(input())

t_list = [0]
x_list = [0]
y_list = [0]
for d in range(n):
    t, x, y = map(int, input().split())
    t_list.append(t)
    x_list.append(x)
    y_list.append(y)

def canTravel():
    for i in range(n):
        dt = t_list[i+1] - t_list[i]
        dist = abs(x_list[i+1] - x_list[i]) + abs(y_list[i+1] - y_list[i])

        if dist > dt:
            return False
        
        if dt % 2 != dist % 2:
            return False
    
    return True

print("Yes" if canTravel() else "No")

File: cpp/beginners_selection/abc086c.cpp
// https://atcoder.jp/contests/abs/tasks/abc085_c

#include<iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    int t[n+1], x[n+1], y[n+1];
    t[0] = 0;
    x[0] = 0;
    y[0] = 0;
    for (int i=1; i<=n; i++) {
        cin >> t[i] >> x[i] >> y[i];
    }

    bool available = true;
    for (int i=1; i<=n; i++) {
        int dt = t[i] - t[i-1];
        int d = abs((x[i]+y[i]) - (x[i-1]+y[i-1]));

        if (dt < d) {
            available = false;
            break;
        }

        int dtr = dt % 2;
        int dr = d % 2;
        if ((dtr == 0 && dr == 0) || (dtr == 1 && dr == 1)) {
            continue;
        } else {
            available = false;
            break;
        }
    }

    string res = available ? "Yes" : "No";
    cout << res << endl;
    return 0;
}
